presto-docs/src/main/sphinx/sql/select.rst

```diff
@@ -577,32 +578,99 @@ output expressions:
 
 Each expression may be composed of output columns or it may be an ordinal
 number selecting an output column by position (starting at one). The
-``ORDER BY`` clause is evaluated as the last step of a query after any
-``GROUP BY`` or ``HAVING`` clause. The default null ordering is ``NULLS LAST``,
-regardless of the ordering direction.
+``ORDER BY`` clause is evaluated after any ``GROUP BY`` or ``HAVING`` clause
+and before any ``OFFSET``, ``LIMIT`` or ``FETCH FIRST`` clause.
+The default null ordering is ``NULLS LAST``, regardless of the ordering direction.
 
-LIMIT Clause
-------------
+.. _offset-clause:
 
-The ``LIMIT`` clause restricts the number of rows in the result set.
-``LIMIT ALL`` is the same as omitting the ``LIMIT`` clause.
-The following example queries a large table, but the limit clause restricts
-the output to only have five rows (because the query lacks an ``ORDER BY``,
+OFFSET Clause
+-------------
+
+The ``OFFSET`` clause is used to discard a number of leading rows
+from the result set:
+
+.. code-block:: none
+
+    OFFSET count [ { ROW | ROWS } ]
+
+If the ``ORDER BY`` clause is present, the ``OFFSET`` clause is evaluated
+over a sorted result set, and the set remains sorted after the
+leading rows are discarded::
+
+    SELECT name FROM nation ORDER BY name OFFSET 22;
+
+.. code-block:: none
+
+          name
+    ----------------
+     UNITED KINGDOM
+     UNITED STATES
+     VIETNAM
+    (3 rows)
+
+Otherwise, it is arbitrary which rows are discarded.
+If the count specified in the ``OFFSET`` clause equals or exceeds the size
+of the result set, the final result is empty.
+
+LIMIT | FETCH FIRST Clause
```

Comment:
Similarly, make this just `FETCH`. Let's also write it as
```
LIMIT or FETCH Clauses
```
Or we could be more like PostgreSQL and keep this as "Limit CLAUSE" with some additional language that also says we support the standard syntax. I like that better, since "LIMIT" is the defacto standard that people are familiar with and will look for in the docs.

Reply:
I think that "FETCH" is not just another syntax for "LIMIT". Currently they differ in minimal allowed row count and default value. When "FETCH ... WITH TIES" is added, which I hope to take care of soon, "FETCH" will need equal mentioning in the docs.